File: ulyp-ui/src/main/java/com/ulyp/ui/code/SourceCode.java
package com.ulyp.ui.code;

public class SourceCode {

    private final String className;
    private final String code;

    public SourceCode(String className, String code) {
        this.className = className;
        this.code = code;
    }

    public String getClassName() {
        return className;
    }

    public String getCode() {
        return code;
    }

    public SourceCode prependToSource(String text) {
        return new SourceCode(this.className, text + this.code);
    }

    public int getLineCount() {
        return code.split("\n").length;
    }

    @Override
    public String toString() {
        return "SourceCode{" +
                "className='" + className + '\'' +
                ", code='" + code + '\'' +
                '}';
    }
}

File: ulyp-ui/src/main/kotlin/com/ulyp/ui/code/SourceCode.kt
package com.ulyp.ui.code

class SourceCode(val className: String, val code: String) {
    fun prependToSource(text: String): SourceCode {
        return SourceCode(className, text + this.code)
    }

    fun getLineCount(): Int {
        return code.split("\n".toRegex()).toTypedArray().size
    }

    override fun toString(): String {
        return "SourceCode{" +
                "className='" + className + '\'' +
                ", code='" + code + '\'' +
                '}'
    }
}